### PR TITLE
Make a lot easier to upload files with http requests

### PR DESCRIPTION
**Pull request by MarianoGnu**
Pull Request sponsored by IMVU

This PR have 2 parts:
1. adds to HTTPRequest class the `request_raw` method
2. Creates the `MultipartFormDataBuffer` class which dumps the fields added to it into a PoolByteArray variable which can be used along with request_raw to upload files.

This is an usage example:

Screen Shot 2019-08-26 at 10 01 04

### CODE REVIEW

File: core/io/http_client.cpp [hunk 1]
```diff
@@ -44,14 +44,14 @@ void MultipartFormDataBuffer::append_string_field(const String &field_name, cons
 
 Error MultipartFormDataBuffer::append_file_field(const String &field_name, const String &file_path, const String &p_custom_mime_type) {
 	Error err;
-	_File *f = memnew(_File);
+	Ref<_File> f(memnew(_File));
```
Comment (merumelu) [inline core/io/http_client.cpp]: This looks odd. Any reason you're not using `FileAccess::open()` directly?

**Comment by akien-mga** [comment]
CC @Faless

**Comment by Faless** [comment]
I understand that uploading small files is probably more common, but still I feel like files should be streamed during request, and not copied all to memory before making the request.
Allocating 100 MiB of RAM to upload a 100 MiB file seems like an overkill...

**Comment by aaronfranke** [comment]
@MarianoGnu  Is this still desired? If so, it needs to be rebased on the latest master branch. Also, the above feedback should be addressed.

Otherwise, abandoned pull requests will be closed in the future as announced [here](https://godotengine.org/article/headsup-vulkan-merged-master-branch-unstable).

**Comment by aaronfranke** [comment]
Closing as abandoned by the author. If this is still desired, feel free to open a new PR, but perhaps discuss with @Faless first.

**Pull request closed without merging**